platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/ExportMigrationContext.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.util.Optional;
+import java.util.function.BiPredicate;
+import java.util.stream.Stream;
+
+/**
+ * The export migration context provides functionality for creating new migration entries and
+ * tracking exported migration entries for a given migratable while processing an export migration
+ * operation.
+ * <p>
+ * <b>
+ * This code is experimental. While this interface is functional
+ * and tested, it may change or be removed in a future version of the
+ * library.
+ * </b>
+ * </p>
+ */
+public interface ExportMigrationContext extends MigrationContext {
+    /**
+     * Creates or retrieves (if already created) a migration entry referenced from the specified system
+     * property to be exported by the corresponding migratable.
+     * <p>
+     * An error will automatically be recorded with the associated migration report if the system
+     * property is not defined or its value is blank.
+     * <p>
+     * <i>Note:</i> The file referenced from the property is assumed to be relative to the current
+     * working directory if not defined as absolute. All paths will automatically be relativized from
+     * ${ddf.home} if located underneath.
+     *
+     * @param name the name of the system property referencing a migration entry to create or retrieve
+     * @return a new migration entry or the existing one if already created for the migration entry
+     * referenced from the specified system property value or empty if the property is not defined or
+     * its value is blank
+     * @throws IllegalArgumentException if <code>name</code> is <code>null</code>
```

Comment:
I am so used to <code>blah</code> that unless I have to change them, I'll keep them as is.